IdentifiabilityODE.mpl

```diff
@@ -557,3 +589,139 @@ CompareDiffVar := proc(dvl, dvr, var_list)
   end if:
   return StringTools[Compare](vr, vl):
 end proc:
+
+
+#===============================================================================
+# Weights
+#===============================================================================
+
+# get rid of unerscore, e.g. x1(t) -> x1_ -> x1
+get_function_name := f -> parse(convert(FunctionToVariable(f), string)[..-2]):
+
+# check if depends on (t)
+is_function:= f->StringTools[Has](convert(f, string), "(t)"):
+idtfm := x->x;
+
+# check if is derivative
+is_diff := f->type(int(f, t), function(name)):
+
+lhs_name := ff -> if convert(ff, string)[-1] = "_" then parse(convert(ff, string)[..-2]) else ff; end if:
+
+#===============================================================================
+GetStateName := proc(state, x_vars, mu)
+#===============================================================================
+  local state_;
+  if state in mu then
+    return state;
+  end if;
+  state_ := parse(cat(StringTools[Join](StringTools[Split](convert(state, string), "_")[..-2], "_"), "_")):
+  if state_ in x_vars then
+    return state_;
+  end if:
+end proc:

```

Comment:
This seem to partially duplicate the function `GetOrderVar`, right? In that case I suggest to merge them and have just one.

Reply:
I understand that. What bothers me is that the convention about how the derivatives encoded is now parsed in two places: here and in `GetOrderVar` meaning that one may, in the future, change only one of them and break down the whole story. How about we replace here the line 617 by application of `GetOrderVar`?

Reply:
It ended up being a wrapper around `GetOrderVar` but now `GetStateName` only relies on one thing. I think this way your concern is addressed. I don't think we should get rid of `GetStateName` completely.